### PR TITLE
Persist grid list view preference

### PR DESCRIPTION
**Pull request by wcjr**

### CODE REVIEW

File: src/views/Requests.svelte [hunk 1]
```diff
@@ -21,24 +20,11 @@ let showAsList = false
...
-  updateQueryString({ list: null })
-
-  viewedRequestsAsList()
-}
 </script>
 
 <FilteredDisplay title="Requests" filter={requestFilter} items={$requests}>
   <div slot="toggles">
-    <GridListToggle on:list={viewAsList} on:grid={viewAsGrid} {showAsList} buttonCssClass="my-1 mx-0" />
+    <GridListToggle on:list={() => showAsList = true} on:grid={() => showAsList = false} class="my-1 mx-0" />
```
Comment (forevermatt) [inline src/views/Requests.svelte]: I'm a little concerned that we're coupling the GridListToggle too tightly to requests. Initially, it was intended merely as a toggle, reusable wherever is convenient. Now that it's internally storing the user's preference for whether to use the grid or list view, that separation of "Tell me what state to show, and I'll tell you when the user selects 'list' or 'grid'" now seems a little muddied. 

A related question is about this `showAsList` here in Requests.svelte: it's being set, but I don't see anywhere that it's used any more... or am I simply overlooking it?

All that said, we could very well make this Request-specific if we so choose. Feel free to ping me and we can discuss (at your convenience).

Comment (wcjr) [reply src/views/Requests.svelte]: https://github.com/silinternational/wecarry-ui/pull/152/commits/a176bb50c33a09f5e25a2edeb96cb1246158fc33

File: src/components/GridListToggle.svelte [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
-import { createEventDispatcher } from 'svelte'
 import { faThList, faTh } from '@fortawesome/free-solid-svg-icons'
+import { createEventDispatcher } from 'svelte'
 
-export let buttonCssClass = ''
-export let showAsList
+export let choice = 'grid'
 
 const dispatch = createEventDispatcher()
+const classes = `btn btn-sm ${$$props.class}`
```
Comment (forevermatt) [inline src/components/GridListToggle.svelte]: If the view using this was defining the `class` dynamically, would this need to be a `let`? We're not doing so at the moment. However, I wondered if using a const for a value dependent upon an input parameter was a little too strict.

Comment (wcjr) [reply src/components/GridListToggle.svelte]: `$$props.class` is not reactive...should we talk through this a little then?

**Pull request merged**